### PR TITLE
build(deps): bump calcite-colors

### PR DESCRIPTION
**Pull request by jcfranco**
**Related Issue:** N/A

## Summary

Bumps `@esri/calcite-colors` to v6.

https://github.com/Esri/calcite-colors/blob/master/CHANGELOG.md#600---2021-01-28

### CODE REVIEW

**Comment by jcfranco** [comment]
@paulcpederson Do you have any suggestions to replace the following removed vars:

* `--calcite-ui-green-2`
* `--calcite-ui-green-3`
* `--calcite-ui-yellow-2`
* `--calcite-ui-yellow-3`

These are referenced in [`calcite-chip.scss`](https://github.com/Esri/calcite-components/blob/804dfd701735c1ae955e865aaee6bd9d1ccb6c27/src/components/calcite-chip/calcite-chip.scss#L145-L148).

**Comment by julio8a** [comment]
@asangma @macandcheese @bstifle ^

My understanding is that we were going to generate those values via code since they are mainly used for hover and press to avoid having so many vars.

**Comment by paulcpederson** [comment]
I think you could accomplish the same style by using white/black low-opacity background colors for the dismiss button. Something like:

```css
:host([appearance="solid"]) button {
    background-color: rgba(0,0,0,0);               /* idle */
    background-color: rgba(0,0,0,.1);              /* press */
    background-color: rgba(255,255,255,.1); /* hover */
}
```

**Comment by macandcheese** [comment]
Yeah, red yellow and blue weren’t going to have hover or press states (and red only for danger - button). I think chip was the only one that had them and we could also swap to white or dark on hover instead of darkening, but above would work too with overlay.

File: conventions/README.md
Comment (julio8a) [inline conventions/README.md:98]: @bstifle, light is used here.

Comment (julio8a) [reply conventions/README.md:98]: I think we should keep these consistent with the CSS variable nomenclature and not use `-light`

CC'ing @paulcpederson

Comment (julio8a) [reply conventions/README.md:98]: Disregard this thread. It's okay as is.

Comment (paulcpederson) [reply conventions/README.md:98]: Yeah, I think that's just what the sass variable is called now they added light to all the light theme colors

**Comment by paulcpederson** [comment]
Wow screener really coming through on this PR catching a lot of visual regressions.

**Comment by julio8a** [comment]
The screener is awesome! License and PO for that are moving again. 
Is there a way to turn off that orange border, it's really hard to see the difference with that. Especially if it's border changes or missing borders.

**Comment by julio8a** [comment]
Switch

Did the toggle handle move or is the color changes affecting the visual placement, Not vertically centered.

image
image

**Comment by bstifle** [comment]
> Yeah, red yellow and blue weren’t going to have hover or press states (and red only for danger - button). I think chip was the only one that had them and we could also swap to white or dark on hover instead of darkening, but above would work too with overlay.

Yes, we agreed that yellow and green do not need hover or press states. we have no use cases for them, hence the removal

**Comment by eriklharper** [comment]
> Is there a way to turn off that orange border, it's really hard to see the difference with that.

Yes!  In the diff view, click the blue sidebar arrow (
image)
 to reveal the left sidebar.  In the left sidebar click "toggle highlight" and that will turn on or off the orange diff lines.

image

**Comment by jcfranco** [comment]
Chip colors updated!

@paulcpederson I tweaked your snippet to follow an [existing selector for the clear appearance](https://github.com/Esri/calcite-components/blob/804dfd701735c1ae955e865aaee6bd9d1ccb6c27/src/components/calcite-chip/calcite-chip.scss#L189-L197). Let me know if I should restore the idle background color for the dismiss button.

chips

**Comment by paulcpederson** [comment]
@jcfranco if screener is to be believed switches are pretty messed up on this brach:

Screen Shot 2021-02-01 at 10 11 16 AM

couple other problems as well.

**Comment by jcfranco** [comment]
@paulcpederson Found the culprit: https://github.com/Esri/calcite-components/blob/804dfd701735c1ae955e865aaee6bd9d1ccb6c27/src/components/calcite-switch/calcite-switch.scss#L88

`--calcite-ui-border-4` was removed but there's no mention of it under the [`CSS Variables` changelog section](https://github.com/Esri/calcite-colors/blob/master/CHANGELOG.md#css-variables).

Should I use `--calcite-ui-border-3` instead? 

Screen Shot 2021-02-01 at 10 51 21 AM

**Comment by julio8a** [comment]
@bstifle ^

**Comment by bstifle** [comment]
@jcfranco border should be ui-text-3 for gray chip

also, for switch, i need more context, you are showing the active-hover state

**Comment by paulcpederson** [comment]
@jcfranco sorry about that I was writing a changelog for code I didn't change, honestly surprised more isn't wrong 🤣  . I can update the release notes to catch this.

**Comment by paulcpederson** [comment]
OK, updated. You could maybe replace the border-4 references with `--calcite-ui-border-input` which is new and I think specifically meant for input borders?

**Comment by jcfranco** [comment]
@paulcpederson Thanks! Is there an equivalent for `--calcite-ui-border-5`? `calcite-action.scss` has the only reference to it.

**Comment by paulcpederson** [comment]
Seems like the closest value for that one in the new set would be `-calcite-ui-text-3`. Would be slightly darker, though (`#757575` => `#6a6a6a`)

**Comment by bstifle** [comment]
Yes please change border-4 to border-input

**Comment by bstifle** [comment]
@jcfranco Yes please change instances of border-4 to border-input. 

Border 5, please reach out to alan since action is the only thing using it

**Comment by jcfranco** [comment]
> border should be ui-text-3 for gray chip

Screen Shot 2021-02-01 at 11 49 43 AM

> Border 5, please reach out to alan since action is the only thing using it

@asangma I've updated action to use `--calcite-ui-text-3`. Here's a comparison:

### before

actions-before

### now

actions-now

**Comment by paulcpederson** [comment]
@bstifle is that correct? The default chip is now clear with an outline? If so, have we gamed that out in how it impacts all UIs (Thinking of combobox where chips are inside an input). This seems very aggressive to me. Clear looks good, though.

**Comment by bstifle** [comment]
> @bstifle is that correct? The default chip is now clear with an outline? If so, have we gamed that out in how it impacts all UIs (Thinking of combobox where chips are inside an input)? This seems very aggressive to me. Clear looks good, though.

No.

Default should be Solid Gray. Not Clear Gray

image

**Comment by paulcpederson** [comment]
@jcfranco can you update that? Looking at the screener seems like this pr is good to go after that fix.

Would be good to get this in so we can cut a new release. I think until this gets merged several UI's appear broken on `master`

**Comment by jcfranco** [comment]
Updated. I hadn't noticed that I was missing replacements for `--calcite-ui-yellow-2` and `--calcite-ui-green-2`. 
Screen Shot 2021-02-02 at 5 21 08 PM

Screen Shot 2021-02-02 at 5 21 02 PM

I'd be happy to hand this off to one of you UXperts™ if anyone wants to volunteer. I feel this would go much faster that way. 🚀

**Comment by paulcpederson** [comment]
@jcfranco hold off merging. There are a couple small color issues still, going to work on them locally. Is it cool if I push to your branch or do you want me to pr into `jcfranco/bump-calcite-colors`?

**Comment by jcfranco** [comment]
10-4! Sure, feel free to push to this branch.

**Comment by bstifle** [comment]
image
image


clear went back to the wrong color, border should be ui-text-3 and ui-text-3-dark ❤️

**Comment by jcfranco** [comment]
Took another pass at the chip colors. I renamed some local vars to hopefully make it clear on how some color vars are being used.

Screen Shot 2021-02-04 at 9 11 19 AM
Screen Shot 2021-02-04 at 9 11 12 AM

**Comment by paulcpederson** [comment]
OK, I've gone through all the screener changes and checked with @bstifle comparing with the figma designs. I think once this build runs I can accept the rest of the changes and we should be good to go.

**Comment by jcfranco** [comment]
Awesome, thanks!

**Review by paulcpederson — APPROVED** [review]

**Pull request merged**
